### PR TITLE
Update security docs

### PR DESCRIPTION
**Pull request by jfoote**
This PR makes some minor updates to the project security docs:

- Move existing SECURITY.md to docs/src
    - This renders correctly in the github repo view; I assume it will work for the docs site as well
- Update SECURITY.md issue-reporting and advisory guidance
    - I _think_ this will automatically be used as a Github Issue template after this is merged but I couldn't find docs to confirm that; if you know otherwise please LMK

### CODE REVIEW

**Comment by jfoote** [comment]
N.b. even if this is automatically used as a Github Issue Template, we may need to do [basic configuration of Github Issue Templates](https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository) so that the security policy shows up in the Github Issue reporting workflow.

File: docs/src/Security.md [hunk 1]
```diff
@@ -1 +1,74 @@
...
+
+## Security model
+
+The Lucet project aims to provide support for secure execution of untrusted code. The project does not provide a complete secure sandbox framework at this time; security is achieved through a combination of Lucet-supplied security controls and user-supplied security controls.
+
+At a high level, this jointly-constructed security architecture aims to prevent untrusted input, data, and activity from compromising the security of trusted components. It also aims to prevent an untrusted actor from compromising the security (e.g. data and activity) of another untrusted actor. For example, one user of a Lucet embedding should not be able to affect the security of another user of the same Lucet embedding.
+
+Some security requirements for the Lucet project have not been implemented yet. See the remainder of this document as well as [project Github issues](https://github.com/bytecodealliance/lucet/issues) for more information. Note that even when lucet project security goals have been met, overall system security requirements will vary by embedding.
```
Comment (katie-martin-fastly) [inline docs/src/Security.md]:
```suggestion
Some security requirements for the Lucet project have not been implemented yet. See the remainder of this document as well as [project Github issues](https://github.com/bytecodealliance/lucet/issues) for more information. Note that even when Lucet project security goals have been met, overall system security requirements will vary by embedding.
```

File: docs/src/Security.md [hunk 2]
```diff
@@ -1 +1,74 @@
...
+
+The Lucet security model can be summarized via two simplified execution scenarios: compiling/loading of sandboxed guest code and execution of untrusted guest programs. These scenarios are described in terms of the following levels.
+
+- Trusted: refers to code, processes, or inputs that are fully trusted and generally controlled by the administrator of a system that runs or embeds Lucet components.
+- Untrusted: refers to code, processes, or inputs that are completely untrusted and generally supplied by a third party. For example, user-supplied WASM code is untrusted.
```
Comment (katie-martin-fastly) [inline docs/src/Security.md]:
```suggestion
- Untrusted: refers to code, processes, or inputs that are completely untrusted and generally supplied by a third party. For example, user-supplied Wasm code is untrusted.
```

**Comment by acfoltzer** [comment]
Can we also move the assets to the docs directory, if they're no longer being referenced at the top level?

**Comment by jfoote** [comment]
Thank you @katie-martin-fastly and @acfoltzer for the feedback! The requested changes have been made. It looks like I somehow dismissed the prior approval when I committed over it -- could you 👍 again (assuming all looks good)?

**Pull request merged**